crates/sui-indexer/src/indexer_reader.rs

```diff
@@ -288,3 +485,27 @@ impl sui_types::storage::ObjectStore for IndexerReader {
             .map_err(|e| sui_types::error::SuiError::GenericStorageError(e.to_string()))
     }
 }
+
+impl move_bytecode_utils::module_cache::GetModule for IndexerReader {
+    type Error = IndexerError;
+
+    type Item = move_binary_format::CompiledModule;
+
+    fn get_module_by_id(
+        &self,
+        id: &move_core_types::language_storage::ModuleId,
+    ) -> Result<Option<Self::Item>, Self::Error> {
+        let package_id = ObjectID::from(*id.address());
+        self.get_package(&package_id)?
+            .map(|package| {
+                package
+                    .deserialize_module(
+                        &id.name().to_owned(),
+                        move_binary_format::file_format_common::VERSION_MAX,
+                        true,
+                    )
+                    .map_err(Into::into)
+            })
+            .transpose()
```

Comment:
In my PR I have a TODO to cache the deserialized module assuming it's not cheap to do it over and over again for hot modules.